pubnub-gson/src/main/java/com/pubnub/internal/endpoints/DeleteMessagesImpl.java

```diff
@@ -0,0 +1,37 @@
+package com.pubnub.internal.endpoints;
+
+import com.pubnub.api.PubNubException;
+import com.pubnub.api.builder.PubNubErrorBuilder;
+import com.pubnub.api.endpoints.DeleteMessages;
+import com.pubnub.api.endpoints.remoteaction.ExtendedRemoteAction;
+import com.pubnub.api.models.consumer.history.PNDeleteMessagesResult;
+import com.pubnub.internal.PubNubCore;
+import lombok.Setter;
+import lombok.experimental.Accessors;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Setter
+@Accessors(chain = true, fluent = true)
+public class DeleteMessagesImpl extends DelegatingEndpoint<PNDeleteMessagesResult> implements DeleteMessages {
+    private List<String> channels = new ArrayList<>();
+    private Long start;
+    private Long end;
+
+    public DeleteMessagesImpl(PubNubCore pubnub) {
+        super(pubnub);
+    }
+
+    @Override
+    protected ExtendedRemoteAction<PNDeleteMessagesResult> createAction() {
```

Comment:
I see that:
In DeleteMessagesImpl.java  createAction returns ExtendedRemoteAction<PNDeleteMessagesResult>
In FetchMessagesImpl.java createAction returns FetchMessagesEndpoint
In GetStateImpl.java createAction returns com.pubnub.internal.EndpointCore<?, PNGetStateResult>

Why there are those discrepancies? Can it be unified?

Reply:
I think you fixed all of those, right? if yes, please resolve

Reply:
Yes, I did.